pkg/actuators/machine/instances.go

```diff
@@ -94,28 +94,14 @@ func getSecurityGroupsIDs(securityGroups []providerconfigv1.AWSResourceReference
 	return securityGroupIDs, nil
 }
 
-func getSubnetIDs(subnet providerconfigv1.AWSResourceReference, availabilityZone string, client awsclient.Client) ([]*string, error) {
+func getSubnetIDs(subnet providerconfigv1.AWSResourceReference, client awsclient.Client) ([]*string, error) {
```

Comment:
Ignoring az here and keeping it here https://github.com/openshift/cluster-api-provider-aws/pull/187/files#diff-24a7c87e11cc6413298ddaaa7e7abda0R274
Open up a window for inconsistencies, e.g the filters give you a subnetid which does not exist in the given az.
I think we should completely drop AZ from the api and explicitely ask consumers rely on filters/subnetID only
cc @bison

Reply:
I agree that just removing the AZ restriction here is problematic.

I also agree that the current `Placement` bit in the API is kind of awkward. I wouldn't be opposed to replacing it, but how to migrate away from it would take some thought and planning.

Reply:
> Open up a window for inconsistencies, e.g the filters give you a subnetid which does not exist in the given az.
1. Even current code i.e with az filter does not help in ^ case. In actuator, we are not dropping filters set by installer and therefore, && result of the filters in mentioned case will be an empty set, no subnet.
2. Since `Placement` and `subnet` in the providerconfig are being set by the same entity, installer, it should be the responsibility of installer to not set inconsistent values. We(Actuator) should not be expected to bring consistency into the inconsistencies set up by installer.

Reply:
Just a general point: I don't think we should develop the actuator with the assumption that the OpenShift installer is the only user. The actuator should absolutely take reasonable measures to protect against inconsistent input.

Reply:
1 - It helps since is adding an additional filter according with the user inputs
2 - This is modeling an API that can be used by consumers in a meaningful manner, installer is one consumer. I agree we should rely on user inputs, I don't think az is bringing any additional value over subnet filters/id. I think we should ask consumers to rely on setting their right filters/id and eventually drop az and region

Reply:
> The actuator should absolutely take reasonable measures to protect against inconsistent input.

actuator should have logic to catch inconsistencies and should fail/return with proper hints about inconsistencies. If inconsistent configuration is received, i doubt actuator can really decide what is consistent.